src/uct/ib/base/ib_iface.c

```diff
@@ -983,6 +1065,29 @@ static ucs_status_t uct_ib_iface_init_gid_info(uct_ib_iface_t *iface,
     return status;
 }
 
+static void uct_ib_iface_set_path_mtu(uct_ib_iface_t *iface,
+                                      const uct_ib_iface_config_t *config)
+{
+    enum ibv_mtu port_mtu = uct_ib_iface_port_attr(iface)->active_mtu;
+    uct_ib_device_t *dev  = uct_ib_iface_device(iface);
+
+    /* MTU is set by user configuration */
+    if (config->path_mtu != UCT_IB_MTU_DEFAULT) {
+        iface->config.path_mtu = (enum ibv_mtu)(config->path_mtu +
+                                                (IBV_MTU_512 - UCT_IB_MTU_512));
```

Comment:
can you explain this please?

Reply:
this is a casting from one enum to other, probably not the safest way (moved as is)